src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -347,7 +389,41 @@ - (void)_handleEventReport:(NSArray<NSDictionary<NSString *, id> *> *)eventRepor
 {
     os_unfair_lock_lock(&self->_lock);
 
-    // first combine with previous unreported events, if they exist
+    NSDate * oldEstimatedStartTime = _estimatedStartTime;
+    for (NSDictionary<NSString *, id> * eventDict in eventReport) {
+        // Whenever a StartUp event is received, reset the estimated start time
+        MTREventPath * eventPath = eventDict[MTREventPathKey];
+        BOOL isStartUpEvent = (eventPath.cluster.unsignedLongValue == MTRClusterIDTypeBasicInformationID)
+            && (eventPath.event.unsignedLongValue == MTRClusterBasicEventStartUpID);
```

Comment:
```suggestion
            && (eventPath.event.unsignedLongValue == MTREventIDTypeClusterBasicInformationEventStartUpID);
```

I assume getting `unsignedLongValue` is preferable to `[eventPath.event isEqualToNumber:@(MTRClusterBasicEventStartUpID)]`?

Reply:
It's minor difference, but this avoids making an object just to compare and throw away..